setuptools/monkey.py

```diff
@@ -152,13 +153,13 @@ def patch_for_msvc_specialized_compiler():
     Patch functions in distutils to use standalone Microsoft Visual C++
     compilers.
     """
-    # import late to avoid circular imports on Python < 3.5
-    msvc = import_module('setuptools.msvc')
-
-    if platform.system() != 'Windows':
+    if not get_platform().startswith('win'):
         # Compilers only availables on Microsoft Windows
         return
 
+    # import late to avoid circular imports on Python < 3.5
```

Comment:
The issue will only occur if `msvc9compiler` thinks that Python was built with VC 6.0 or similar, which it defaults to when Python is built with mingw since this function [Lib/distutils/msvc9compiler.py:169-192](https://github.com/python/cpython/blob/778928b0c7aa438c282727535814d73df850693a/Lib/distutils/msvc9compiler.py#L169-L192) won't find the MSVC version in the `sys.version`. (Yes it's *"parsing"* `sys.version`... 🙄)